README.md

```diff
@@ -142,11 +142,11 @@ for i in Range::new(1usize, 4, 1) {
 }
 ```
 
-Straighforward stuff. This does demonstrate a couple important characteristics of iterators, though.
+Straightforward stuff. This does demonstrate a couple of important characteristics of iterators, though.
 
 **Laziness**
 
-Iterators are lazy, they don't do anything unless _polled_.  
+Iterators are lazy; they don't do anything unless _polled_.  
 In this case the for loop is doing the polling, as it desugars to something along these lines:
```

Comment:
Another comma splice.

Reply:
That could also be a colon, and perhaps should be, since the right elucidates the left.